### DISCUSSION TITLE
Support for multiple inverters in the same network/facility?

### DISCUSSION THREAD

**Discussion by hanktos** [mletenay/home-assistant-goodwe-inverter#135] [Q&A]
I have been running this integration since the beginning 2 years ago and it has been a great pleasure!

I am currently completing my installation with two more inverters (ET) so I will have a total of 3 inverters (ET)
The question I asked myself is how this integration handles multiple inverters and how I read/write data from/to them all.

Is there support for this in the integration or do you have any tips on how I can solve it?
One thought I had if support is missing was to duplicate the integration and create multiple instances of the same integration 01, 02, 03, etc.
Or maby duplicate "slave" HA instances with native integrations per HA instance and collect data to master HA instance?

Other tips and ideas?

**Comment by mletenay** [comment]
The integration should support multliple inverters, there is at least one user who runs 2 ES inverters.
You just need to add the integration several times, with different inverter IPs, it should then create 3 devices and 3 sets of entities.

**Reply by JoLambinon** [reply]
I have tried to add a second ES inverter by adding the integration again with the second IP address but get an unknown error from HA when I attempt this and not allowed to add the integration twice

**Comment by rhoracek** [comment]
Hello, please, add the option of integrating 2 inverters, when adding the second inverter, the integration ends with an error. Thank you

**Reply by mletenay** [reply]
As written above, the integration does support multiple inverters.
You have to attach log files to investigate potential problems

**Comment by crispe** [comment]
I had 3 goodwe inverters connected simultaneously, works perfectly.
